src/js/base/ripe.js

```diff
@@ -56,7 +56,7 @@ ripe.RipeBase = function(brand, model, options = {}) {
  */
 ripe.Ripe.fromStructure = async function(structure, safe = true, options = {}) {
     const instance = new ripe.Ripe(options);
-    await instance.setStructure(structure, safe);
+    await instance.setStructure(structure, safe, options);

```

Comment:
I don't think you need this here, given that in the line above you're already passing the options